src/test/java/build/buildfarm/instance/shard/RedisShardSubscriptionTest.java

```diff
@@ -107,7 +108,6 @@ public void onSubscribe(String channel, int subscribedChannels) {
     Thread thread = new Thread(subscription);
     thread.start();
 
-    // FIXME if we stop before we are subscribed, this hangs
     subscribed.get();
```

Comment:
This is now (hopefully) unnecessary complexity. Remove subscribed as it should be inconsequential.

Reply:
This is required still. The first attempt to subscribe is set up to fail so that onReset will be called, so we need to wait for this future or else stop will be called before the subscription completes and we'll get a jedis exception.